Review

Romancompany opened the pull request:
https://github.com/Romancompany/stellar-burgers.git

Romancompany committed:
старт

Romancompany committed:
работает оформление заказа

Romancompany committed:
конструктор, лента заказов, история заказов

Romancompany committed:
доработал orderByNumberSlice

Romancompany committed:
добавил protected-route.tsx

Romancompany committed:
закончил КП11

SergeyK2004 commented on src/components/app/app.tsx:
```diff
@@ -1,14 +1,105 @@
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  useEffect(() => {
+    dispatch(checkUserAuth());
```
Надо исправить:
При открытии сайта, даже если пользователь был ранее авторизован, сайт открывается без авторизации пользователя.

SergeyK2004 commented on src/components/burger-constructor-element/burger-constructor-element.tsx:
```diff
@@ -1,14 +1,20 @@
 import { FC, memo } from 'react';
 import { BurgerConstructorElementUI } from '@ui';
 import { BurgerConstructorElementProps } from './type';
+import { useDispatch } from '../../services/store';
```
Отлично:
Правильно, что используете типизированные хуки

SergeyK2004 commented on src/components/ui/app-header/app-header.tsx:
```diff
@@ -7,28 +7,53 @@ import {
         <>
           <BurgerIcon type={'primary'} />
-          <p className='text text_type_main-default ml-2 mr-10'>Конструктор</p>
+          <NavLink
```
Отлично:
Использование NavLink в навигационном меню

SergeyK2004 commented on src/components/burger-ingredient/burger-ingredient.tsx:
```diff
@@ -1,14 +1,28 @@
+      if (ingredient.type === ('bun' as TTabMode)) {
+        dispatch(setBunIngredient(ingredient));
+      } else {
+        const id = nanoid();
```
Отлично:
Хорошо ,что id вычисляете не в редьюсере.

SergeyK2004 reviewed (changes requested):
Здравствуйте. 

Работа проделана огромная

Что сделано отлично:
Отлично сделана логика проверки, был ли пользователь на сайте, и показывать ли страницу с деталями, а не модальное окно. Весь заявленный функционал реализован. Здорово, что выделяется активная страница в главном меню. Бургер корректно собирается с возможностью менять ингредиенты местами.

Что нужно исправить:

* https://disk.yandex.ru/d/Haaf79y4qB8Ecg  Заголовок формы некорректный. Отсутствует статус заказа. В заказе отсутствуют булки. Сверьтесь с макетом. Проверьте так-же отображение заказа в отдельной странице.
* При открытии модального окна, фон за ним не должен быть пустым. Модальное окно открывается на фоне той странице, в которой его открыли.
* Если пользователь уже был на сайте авторизован, то при повторном открытии сайта он должен быть авторизован автоматически.
* Ингредиенты и заказы, открываемые в отдельной странице по прямой ссылке, должны открываться как страница сайта, а не модельное окно.
* https://disk.yandex.ru/d/d7aA5B_yCzCQ9A Не корректно выделяется активный пункт меню. Обратите внимание, что иконки в меню должны так-же изменять свое отображение в зависимости от активности пункта меню. Сверьтесь с макетом.

Можно лучше:

* Будет здорово, если ингредиенты в заказе можно будет менять местами. Для этого предусмотрены кнопки слева от названия ингредиента.

Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка F12) перед отправкой на ревью. 
Напоминаю, что работа может быть принята только после исправления всех критических замечаний Нужно исправить.


Удачного рефакторинга кода.

Romancompany committed:
исправил замечания:
отсутствует статус заказа
отсутсвтуют булки в заказе
повторная открытие сайта с авторизованным пользователем
иконка в активном пункте

Romancompany committed:
исправил все замечания

Romancompany committed:
сверил с макетом

Romancompany committed:
сделал up и down инградиентов в конструкторе бургера

SergeyK2004 reviewed (approved):
Отлично. Практически все замечания исправлены. Осталось только одно:

* https://disk.yandex.ru/d/pIyhPGSwgZcD0A Не корректно выделяется активный пункт меню. Сверьтесь с макетом.

Исправьте это самостоятельно. Ваша работа принята. Желаю успехов в дальнейшем обучении.
Обратите внимание, что сейчас вам нужно смержить ветку review в main и после этого, при выполнении проектной работы следующего спринта создать новую ветку review, в которой вы будете покрывать приложение тестами. Старую ветку review после слияния можно удалить, чтобы она не мешала вам создавать новую.

Romancompany committed:
убрал подчеркивание на тексте активной ссылку верхнего меню
